fraud/bad_encoding.go

```diff
@@ -173,3 +175,31 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 
 	return nil
 }
+
+// OnBEFP listens to Bad Encoding Fraud Proof and stops services immediately if it is received.
+func OnBEFP(ctx context.Context, s Subscriber, stop func(context.Context) error) {
+	subscription, err := s.Subscribe(BadEncoding)
+	defer func() {
+		if err != context.Canceled {
+			if stopErr := stop(ctx); err != nil {
+				log.Warn(stopErr)
+			}
+		}
+	}()
```

Comment:
```suggestion
func OnBEFP(ctx context.Context, s Subscriber, stop func(context.Context) (err error) {
	defer func() {
		if err == context.Canceled {
			return
		}
		
		stopErr := stop(ctx)
		if stopErr != nil {
			log.Warn(stopErr)
		}
	}()
	
	subscription, err := s.Subscribe(BadEncoding)
```

Reply:
* This is a good case to define a global named return err and make defer for it.
* Don't do nested if
* there was a bug where stopErr was not checked in if before log.Warn